File: event-service/src/main/java/com/urbanShows/eventService/dto/EventPage.java
package com.urbanShows.eventService.dto;

import java.util.List;

import com.urbanShows.eventService.constant.TableConfig;

import lombok.Data;

@Data
public class EventPage {

	private int totalRecords;
	private int filteredRecords;
	private int totalPages;
	private int currentPage;
	private int rowStartIndex;
	private int rowEndIndex;
	private List<Integer> displayPagesIndex;
	private int recordsPerPage = TableConfig.PAGE_SIZE;

}

File: admin-dashboard/src/models/EventPage.ts
import { SearchFilter } from './SearchFilter';

export interface EventPage {
  totalRecords: number;
  filteredRecords: number;
  totalPages: number;
  currentPage: number;
  rowStartIndex: number;
  rowEndIndex: number;
  recordsPerPage: number;
  displayPagesIndex: number[];
}
